src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const storyOfActions = [];
+  const changeableObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(changeableObj, action.extraData);
+        storyOfActions.push(Object.assign({}, changeableObj, action.extraData));
```

Comment:
```suggestion
        storyOfActions.push({ ...changeableObj });
```
Don't repeat yourself. You can write this statement only once, at the end of the iteration.